Sólo en old/data: PEAR
Sólo en old/data: Structures_Graph
diff -urp old/data/.svn/entries new/data/.svn/entries
--- old/data/.svn/entries	2008-05-22 16:25:17.000000000 -0430
+++ new/data/.svn/entries	2008-05-22 16:25:42.000000000 -0430
@@ -1,7 +1,7 @@
 8
 
 dir
-9625
+13041
 http://cidesa.svn:8080/repositorio/web/php/siga/trunk/varios/data
 http://cidesa.svn:8080/repositorio
 
Sólo en old/data/symfony: bin
Sólo en old/data/symfony: config
Sólo en old/data/symfony: data
diff -urp old/data/symfony/generator/sfPropelAdmin/default/skeleton/actions/actions.class.php new/data/symfony/generator/sfPropelAdmin/default/skeleton/actions/actions.class.php
--- old/data/symfony/generator/sfPropelAdmin/default/skeleton/actions/actions.class.php	2008-05-22 16:25:18.000000000 -0430
+++ new/data/symfony/generator/sfPropelAdmin/default/skeleton/actions/actions.class.php	2008-05-22 16:25:42.000000000 -0430
@@ -10,4 +10,275 @@
  */
 class ##MODULE_NAME##Actions extends auto##MODULE_NAME##Actions
 {
+
+  private $coderr = -1;
+
+  public function executeEdit()
+  {
+   /*Falta parametros para que funcione los mensaje de error
+   *Consultar con luis...
+   **/
+    parent::executeEdit();
+
+  }
+
+  public function configGrid($reg = array(),$regelim = array())
+  {
+    $this->regelim = $regelim;
+
+    if(!count($reg)>0)
+    {
+      // Aquí va el código para traernos los registros que contendrá el grid
+      $reg = array();
+      // Aquí va el código para generar arreglo de configuración del grid
+	  $this->obj = array();
+    }
+
+    // Insertar el criterio de la busqueda de registros del Grid
+    // Por ejemplo:
+
+    // $c = new Criteria();
+    // $c->add(CaartaocPeer::AJUOC ,$this->caajuoc->getAjuoc());
+    // $reg = CaartaocPeer::doSelect($c);
+
+    // De esta forma se carga la configuración del grid de un archivo yml
+    // y se le pasa el parámetro de los registros encontrados ($reg)
+    //                                                                            /nombreformulario/
+    // $this->obj = Herramientas::getConfigGrid(sfConfig::get('sf_app_module_dir').'/formulario/'.sfConfig::get('sf_app_module_config_dir_name').'/grid_caartaoc',$reg);
+
+    // Si no se quiere cargar la configuración del grid de un .yml, sedebe hacer a pie.
+    // Por ejemplo:
+
+    /*
+    // Se crea el objeto principal de la clase OpcionesGrid
+    $opciones = new OpcionesGrid();
+    // Se configuran las opciones globales del Grid
+    $opciones->setEliminar(true);
+    $opciones->setTabla('Caartalm');
+    $opciones->setAnchoGrid(1150);
+    $opciones->setTitulo('Existencia por Almacenes');
+    $opciones->setHTMLTotalFilas(' ');
+    // Se generan las columnas
+    $col1 = new Columna('Cod. Almacen');
+    $col1->setTipo(Columna::TEXTO);
+    $col1->setEsGrabable(true);
+    $col1->setAlineacionObjeto(Columna::CENTRO);
+    $col1->setAlineacionContenido(Columna::CENTRO);
+    $col1->setNombreCampo('codalm');
+    $col1->setCatalogo('cadefalm','sf_admin_edit_form','2');
+    $col1->setAjax(2,2);
+
+    $col2 = new Columna('Descripción');
+    $col2->setTipo(Columna::TEXTO);
+    $col2->setAlineacionObjeto(Columna::IZQUIERDA);
+    $col2->setAlineacionContenido(Columna::IZQUIERDA);
+    $col2->setNombreCampo('codalm');
+    $col2->setHTML('type="text" size="25" disabled=true');
+
+    // Se guardan las columnas en el objetos de opciones
+    $opciones->addColumna($col1);
+    $opciones->addColumna($col2);
+
+    // Se genera el arreglo de opciones necesario para generar el grid
+    $this->obj = $opciones->getConfig($per);
+     */
+
+
+  }
+
+  public function executeAjax()
+  {
+
+    $codigo = $this->getRequestParameter('codigo','');
+    // Esta variable ajax debe ser usada en cada llamado para identificar
+    // que objeto hace el llamado y por consiguiente ejecutar el código necesario
+    $ajax = $this->getRequestParameter('ajax','');
+
+    // Se debe enviar en la petición ajax desde el cliente los datos que necesitemos
+    // para generar el código de retorno, esto porque en un llamado Ajax no se devuelven
+    // los datos de los objetos de la vista como pasa en un submit normal.
+
+    switch ($ajax){
+      case '1':
+        // La variable $output es usada para retornar datos en formato de arreglo para actualizar
+        // objetos en la vista. mas informacion en
+        // http://201.210.211.26:8080/www/wiki/index.php/Agregar_Ajax_para_buscar_una_descripcion
+        $output = '[["","",""],["","",""],["","",""]]';
+        break;
+      default:
+        $output = '[["","",""],["","",""],["","",""]]';
+    }
+
+    // Instruccion para escribir en la cabecera los datos a enviar a la vista
+    $this->getResponse()->setHttpHeader("X-JSON", '('.$output.')');
+
+    // Si solo se va usar ajax para actualziar datos en objetos ya existentes se debe
+    // mantener habilitar esta instrucción
+    return sfView::HEADER_ONLY;
+
+    // Si por el contrario se quiere reemplazar un div en la vista, se debe deshabilitar
+    // por supuesto tomando en cuenta que debe existir el archivo ajaxSuccess.php en la carpeta templates.
+
+  }
+
+  public function save##MODEL_CLASS##($##MODEL_CLASS##)
+  {
+    $coderr = -1;
+
+    // habilitar la siguiente línea si se usa grid
+    //$grid=Herramientas::CargarDatosGrid($this,$this->obj);
+
+    try {
+
+      // Modificar la siguiente línea para llamar al método
+      // correcto en la clase del negocio, ej:
+      // $coderr = Compras::salvarAlmaujoc($caajuoc,$grid);
+
+      // OJO ----> Eliminar esta linea al modificar este método
+      parent::save##MODEL_CLASS##($##MODEL_CLASS##);
+
+      if(is_array($coderr)){
+        foreach ($coderr as $ERR){
+          $err = Herramientas::obtenerMensajeError($ERR);
+          $this->getRequest()->setError('',$err);
+          $this->ActualizarGrid();
+        }
+      }elseif($coderr!=-1){
+        $err = Herramientas::obtenerMensajeError($coderr);
+        $this->getRequest()->setError('',$err);
+        $this->ActualizarGrid();
+      }
+
+    } catch (Exception $ex) {
+
+      $coderr = 0;
+      $err = Herramientas::obtenerMensajeError($coderr);
+      $this->getRequest()->setError('',$err);
+
+    }
+
+
+  }
+
+
+  public function delete##MODEL_CLASS##($##MODEL_CLASS##)
+  {
+
+    $coderr = -1;
+
+    // habilitar la siguiente línea si se usa grid
+    //$grid=Herramientas::CargarDatosGrid($this,$this->obj);
+
+    try {
+
+      // Modificar la siguiente línea para llamar al método
+      // correcto en la clase del negocio, ej:
+      // $coderr = Compras::EliminarAlmaujoc($caajuoc,$grid);
+
+      // OJO ----> Eliminar esta linea al modificar este método
+      parent::delete##MODEL_CLASS##($##MODEL_CLASS##);
+
+      if(is_array($coderr)){
+        foreach ($coderr as $ERR){
+          $err = Herramientas::obtenerMensajeError($ERR);
+          $this->getRequest()->setError('',$err);
+          $this->ActualizarGrid();
+        }
+      }elseif($coderr!=-1){
+        $err = Herramientas::obtenerMensajeError($coderr);
+        $this->getRequest()->setError('',$err);
+        $this->ActualizarGrid();
+      }
+
+
+    } catch (Exception $ex) {
+
+      $coderr = 0;
+      $err = Herramientas::obtenerMensajeError($coderr);
+      $this->getRequest()->setError('',$err);
+
+    }
+
+  }
+
+  public function validateEdit()
+  {
+    $resp=-1;
+
+    // Se deben llamar a las funciones necesarias para cargar los
+    // datos de la vista que serán usados en las funciones de validación.
+    // Por ejemplo:
+
+    if($this->getRequest()->getMethod() == sfRequest::POST){
+
+      // $this->caajuoc = $this->getCaajuocOrCreate();
+      // $this->##MODEL_CLASS##= $this->get##MODEL_CLASS##OrCreate();
+      // $this->configGrid();
+      // $grid = Herramientas::CargarDatosGrid($this,$this->obj);
+
+      // Aqui van los llamados a los métodos de las clases del
+      // negocio para validar los datos.
+      // Los resultados de cada llamado deben ser analizados por ejemplo:
+
+      // $resp = Compras::validarAlmajuoc($this->caajuoc,$grid);
+      // o
+       //Para que el codigo no se pueda cambiar al editar el registro:
+       //$this->tstipmov= $this->getTstipmovOrCreate();
+       //$tstipmov = $this->getRequestParameter('tstipmov');
+       //$valor = $tstipmov['codtip'];
+       //$campo='codtip';
+
+       //$resp=Herramientas::ValidarCodigo($valor,$this->tstipmov,$campo);
+
+      // al final $resp es analizada en base al código que retorna
+      // Todas las funciones de validación y procesos del negocio
+      // deben retornar códigos >= -1. Estos código serám buscados en
+      // el archivo errors.yml en la función handleErrorEdit()
+
+      if($resp!=-1){
+        $this->coderr = $resp;
+        return false;
+      } else return true;
+
+    }else return true;
+
+
+
+  }
+
+  public function handleErrorEdit()
+  {
+    $this->labels = $this->getLabels();
+
+    //Colocar la Primera letra en minuscula
+    $this->##MODEL_CLASS##= $this->get##MODEL_CLASS##OrCreate();
+    $this->update##MODEL_CLASS##FromRequest();
+
+
+    if($this->getRequest()->getMethod() == sfRequest::POST)
+    {
+      if($this->coderr!=-1){
+        $err = Herramientas::obtenerMensajeError($this->coderr);
+        $this->getRequest()->setError('',$err);
+      }
+    }
+    return sfView::SUCCESS;
+
+  }
+
+  /**
+   * Función para actualziar el grid en el post si ocurre un error
+   * Se pueden colocar aqui los grids adicionales
+   *
+   */
+  public function ActualizarGrid()
+  {
+    $this->configGrid();
+
+    $grid = Herramientas::CargarDatosGrid($this,$this->obj);
+
+    $this->configGrid($grid[0],$grid[1]);
+
+  }
+
 }
diff -urp old/data/symfony/generator/sfPropelAdmin/default/skeleton/actions/.svn/entries new/data/symfony/generator/sfPropelAdmin/default/skeleton/actions/.svn/entries
--- old/data/symfony/generator/sfPropelAdmin/default/skeleton/actions/.svn/entries	2008-05-22 16:25:18.000000000 -0430
+++ new/data/symfony/generator/sfPropelAdmin/default/skeleton/actions/.svn/entries	2008-05-22 16:25:42.000000000 -0430
@@ -1,7 +1,7 @@
 8
 
 dir
-9625
+13041
 http://cidesa.svn:8080/repositorio/web/php/siga/trunk/varios/data/symfony/generator/sfPropelAdmin/default/skeleton/actions
 http://cidesa.svn:8080/repositorio
 
diff -urp old/data/symfony/generator/sfPropelAdmin/default/skeleton/config/generator.yml new/data/symfony/generator/sfPropelAdmin/default/skeleton/config/generator.yml
--- old/data/symfony/generator/sfPropelAdmin/default/skeleton/config/generator.yml	2008-05-22 16:25:18.000000000 -0430
+++ new/data/symfony/generator/sfPropelAdmin/default/skeleton/config/generator.yml	2008-05-22 16:25:42.000000000 -0430
@@ -3,3 +3,9 @@ generator:
   param:
     model_class:      ##MODEL_CLASS##
     theme:            ##THEME##
+
+    edit:
+      actions:
+        _list:        ~
+        _save:        ~
+        _delete:      ~
\ No hay ningún carácter de nueva línea al final del fichero
diff -urp old/data/symfony/generator/sfPropelAdmin/default/skeleton/config/.svn/entries new/data/symfony/generator/sfPropelAdmin/default/skeleton/config/.svn/entries
--- old/data/symfony/generator/sfPropelAdmin/default/skeleton/config/.svn/entries	2008-05-22 16:25:18.000000000 -0430
+++ new/data/symfony/generator/sfPropelAdmin/default/skeleton/config/.svn/entries	2008-05-22 16:25:42.000000000 -0430
@@ -1,7 +1,7 @@
 8
 
 dir
-9625
+13041
 http://cidesa.svn:8080/repositorio/web/php/siga/trunk/varios/data/symfony/generator/sfPropelAdmin/default/skeleton/config
 http://cidesa.svn:8080/repositorio
 
diff -urp old/data/symfony/generator/sfPropelAdmin/default/skeleton/.svn/entries new/data/symfony/generator/sfPropelAdmin/default/skeleton/.svn/entries
--- old/data/symfony/generator/sfPropelAdmin/default/skeleton/.svn/entries	2008-05-22 16:25:18.000000000 -0430
+++ new/data/symfony/generator/sfPropelAdmin/default/skeleton/.svn/entries	2008-05-22 16:25:42.000000000 -0430
@@ -1,7 +1,7 @@
 8
 
 dir
-9625
+13041
 http://cidesa.svn:8080/repositorio/web/php/siga/trunk/varios/data/symfony/generator/sfPropelAdmin/default/skeleton
 http://cidesa.svn:8080/repositorio
 
diff -urp old/data/symfony/generator/sfPropelAdmin/default/skeleton/templates/.svn/entries new/data/symfony/generator/sfPropelAdmin/default/skeleton/templates/.svn/entries
--- old/data/symfony/generator/sfPropelAdmin/default/skeleton/templates/.svn/entries	2008-05-22 16:25:18.000000000 -0430
+++ new/data/symfony/generator/sfPropelAdmin/default/skeleton/templates/.svn/entries	2008-05-22 16:25:42.000000000 -0430
@@ -1,7 +1,7 @@
 8
 
 dir
-9625
+13041
 http://cidesa.svn:8080/repositorio/web/php/siga/trunk/varios/data/symfony/generator/sfPropelAdmin/default/skeleton/templates
 http://cidesa.svn:8080/repositorio
 
diff -urp old/data/symfony/generator/sfPropelAdmin/default/.svn/entries new/data/symfony/generator/sfPropelAdmin/default/.svn/entries
--- old/data/symfony/generator/sfPropelAdmin/default/.svn/entries	2008-05-22 16:25:17.000000000 -0430
+++ new/data/symfony/generator/sfPropelAdmin/default/.svn/entries	2008-05-22 16:25:42.000000000 -0430
@@ -1,7 +1,7 @@
 8
 
 dir
-9625
+13041
 http://cidesa.svn:8080/repositorio/web/php/siga/trunk/varios/data/symfony/generator/sfPropelAdmin/default
 http://cidesa.svn:8080/repositorio
 
Sólo en old/data/symfony/generator/sfPropelAdmin/default/template/actions: actions.class.php
diff -urp old/data/symfony/generator/sfPropelAdmin/default/template/actions/.svn/entries new/data/symfony/generator/sfPropelAdmin/default/template/actions/.svn/entries
--- old/data/symfony/generator/sfPropelAdmin/default/template/actions/.svn/entries	2008-05-22 16:25:17.000000000 -0430
+++ new/data/symfony/generator/sfPropelAdmin/default/template/actions/.svn/entries	2008-05-22 16:25:42.000000000 -0430
@@ -1,7 +1,7 @@
 8
 
 dir
-9625
+13041
 http://cidesa.svn:8080/repositorio/web/php/siga/trunk/varios/data/symfony/generator/sfPropelAdmin/default/template/actions
 http://cidesa.svn:8080/repositorio
 
diff -urp old/data/symfony/generator/sfPropelAdmin/default/template/.svn/entries new/data/symfony/generator/sfPropelAdmin/default/template/.svn/entries
--- old/data/symfony/generator/sfPropelAdmin/default/template/.svn/entries	2008-05-22 16:25:17.000000000 -0430
+++ new/data/symfony/generator/sfPropelAdmin/default/template/.svn/entries	2008-05-22 16:25:42.000000000 -0430
@@ -1,7 +1,7 @@
 8
 
 dir
-9625
+13041
 http://cidesa.svn:8080/repositorio/web/php/siga/trunk/varios/data/symfony/generator/sfPropelAdmin/default/template
 http://cidesa.svn:8080/repositorio
 
Sólo en old/data/symfony/generator/sfPropelAdmin/default/template/templates: _edit_actions.php
Sólo en old/data/symfony/generator/sfPropelAdmin/default/template/templates: _edit_footer.php
Sólo en old/data/symfony/generator/sfPropelAdmin/default/template/templates: _edit_form.php
Sólo en old/data/symfony/generator/sfPropelAdmin/default/template/templates: _edit_header.php
diff -urp old/data/symfony/generator/sfPropelAdmin/default/template/templates/_edit_messages.php new/data/symfony/generator/sfPropelAdmin/default/template/templates/_edit_messages.php
--- old/data/symfony/generator/sfPropelAdmin/default/template/templates/_edit_messages.php	2008-05-22 16:25:18.000000000 -0430
+++ new/data/symfony/generator/sfPropelAdmin/default/template/templates/_edit_messages.php	2008-05-22 16:25:42.000000000 -0430
@@ -1,9 +1,9 @@
 [?php if ($sf_request->hasErrors()): ?]
 <div class="form-errors">
-<h2>[?php echo __('The form is not valid because it contains some errors.') ?]</h2>
+<h2>[?php echo __('There are some errors that prevent the form to validate') ?]</h2>
 <dl>
 [?php foreach ($sf_request->getErrorNames() as $name): ?]
-  <dt>[?php echo __($labels[$name]) ?]</dt>
+  <dt>[?php if ($name!='') echo __($labels[$name]) ?]</dt>
   <dd>[?php echo $sf_request->getError($name) ?]</dd>
 [?php endforeach; ?]
 </dl>
Sólo en old/data/symfony/generator/sfPropelAdmin/default/template/templates: editSuccess.php
Sólo en old/data/symfony/generator/sfPropelAdmin/default/template/templates: _filters.php
Sólo en old/data/symfony/generator/sfPropelAdmin/default/template/templates: _list_actions.php
Sólo en old/data/symfony/generator/sfPropelAdmin/default/template/templates: _list_footer.php
Sólo en old/data/symfony/generator/sfPropelAdmin/default/template/templates: _list_header.php
Sólo en old/data/symfony/generator/sfPropelAdmin/default/template/templates: _list_messages.php
Sólo en old/data/symfony/generator/sfPropelAdmin/default/template/templates: _list.php
Sólo en old/data/symfony/generator/sfPropelAdmin/default/template/templates: listSuccess.php
Sólo en old/data/symfony/generator/sfPropelAdmin/default/template/templates: _list_td_actions.php
Sólo en old/data/symfony/generator/sfPropelAdmin/default/template/templates: _list_td_stacked.php
Sólo en old/data/symfony/generator/sfPropelAdmin/default/template/templates: _list_td_tabular.php
Sólo en old/data/symfony/generator/sfPropelAdmin/default/template/templates: _list_th_stacked.php
Sólo en old/data/symfony/generator/sfPropelAdmin/default/template/templates: _list_th_tabular.php
diff -urp old/data/symfony/generator/sfPropelAdmin/default/template/templates/.svn/entries new/data/symfony/generator/sfPropelAdmin/default/template/templates/.svn/entries
--- old/data/symfony/generator/sfPropelAdmin/default/template/templates/.svn/entries	2008-05-22 16:25:17.000000000 -0430
+++ new/data/symfony/generator/sfPropelAdmin/default/template/templates/.svn/entries	2008-05-22 16:25:42.000000000 -0430
@@ -1,7 +1,7 @@
 8
 
 dir
-9625
+13041
 http://cidesa.svn:8080/repositorio/web/php/siga/trunk/varios/data/symfony/generator/sfPropelAdmin/default/template/templates
 http://cidesa.svn:8080/repositorio
 
diff -urp old/data/symfony/generator/sfPropelAdmin/.svn/entries new/data/symfony/generator/sfPropelAdmin/.svn/entries
--- old/data/symfony/generator/sfPropelAdmin/.svn/entries	2008-05-22 16:25:17.000000000 -0430
+++ new/data/symfony/generator/sfPropelAdmin/.svn/entries	2008-05-22 16:25:42.000000000 -0430
@@ -1,7 +1,7 @@
 8
 
 dir
-9625
+13041
 http://cidesa.svn:8080/repositorio/web/php/siga/trunk/varios/data/symfony/generator/sfPropelAdmin
 http://cidesa.svn:8080/repositorio
 
Sólo en old/data/symfony/generator: sfPropelCrud
diff -urp old/data/symfony/generator/.svn/entries new/data/symfony/generator/.svn/entries
--- old/data/symfony/generator/.svn/entries	2008-05-22 16:25:17.000000000 -0430
+++ new/data/symfony/generator/.svn/entries	2008-05-22 16:25:42.000000000 -0430
@@ -1,7 +1,7 @@
 8
 
 dir
-9625
+13041
 http://cidesa.svn:8080/repositorio/web/php/siga/trunk/varios/data/symfony/generator
 http://cidesa.svn:8080/repositorio
 
Sólo en old/data/symfony: i18n
Sólo en old/data/symfony: modules
Sólo en old/data/symfony: skeleton
diff -urp old/data/symfony/.svn/entries new/data/symfony/.svn/entries
--- old/data/symfony/.svn/entries	2008-05-22 16:25:17.000000000 -0430
+++ new/data/symfony/.svn/entries	2008-05-22 16:25:42.000000000 -0430
@@ -1,7 +1,7 @@
 8
 
 dir
-9625
+13041
 http://cidesa.svn:8080/repositorio/web/php/siga/trunk/varios/data/symfony
 http://cidesa.svn:8080/repositorio
 
Sólo en old/data/symfony: tasks
Sólo en old/data/symfony: web
